packages/@expo/metro-config/src/transform-plugins/import-export-plugin.ts

```diff
@@ -775,183 +478,558 @@ export function importExportPlugin({
-                    REQUIRED: path.scope.generateUidIdentifier(e.file),
-                    KEY: path.scope.generateUidIdentifier('key'),
-                  }),
-                  e.loc
-                )
+          if (state.opts.liveBindings) {
+            const _namespaceForLocal = new Map<string, { namespace: string; name: string }>();
+            const liveExports: t.Statement[] = [];
+            const exportAll: t.Statement[] = [];
+            const imports: t.Statement[] = [];
+            const staticExports: t.Statement[] = [];
+            const defaultStaticExports: t.Statement[] = [];
+
+            const exportedNames: Set<string> = new Set();
+            for (const e of state.exportAllFromAs.values()) {
+              for (const { as } of e) {
+                exportedNames.add(as);
+              }
+            }
+            for (const e of state.exportNamedFrom.values()) {
+              for (const { as } of e) {
+                exportedNames.add(as);
+              }
+            }
+            for (const { as } of state.exportNamed) {
+              exportedNames.add(as);
+            }
+            const exportedNamesIdentifier = path.scope.generateUidIdentifier('_exportedNames');
+            if (state.exportAllFrom.size) {
+              // Generate a static list of exported names to runtime overwrites;
+              liveExports.push(
+                exportedNamesTemplate({
+                  t,
+                  IDENTIFIER: exportedNamesIdentifier.name,
+                  NAMES: [...exportedNames],
+                })
               );
             }
-          });
 
-          state.exportDefault.forEach((e) => {
-            if (e.namespace) {
-              exports.push(
-                withLocation(
-                  liveBindExportTemplate({
-                    REQUIRED: t.identifier(e.namespace),
-                    LOCAL: t.identifier(e.local),
-                    REMOTE: 'default',
-                  }),
-                  e.loc
-                )
-              );
-            } else {
-              body.push(
-                withLocation(
-                  exportTemplate({
-                    LOCAL: t.identifier(e.local),
-                    REMOTE: t.identifier('default'),
-                  }),
-                  e.loc
-                )
-              );
+            for (const [, module] of state.originalImportOrder) {
+              const resolved = resolvePath(module, state.opts.resolve);
+
+              let namespace: t.Identifier | null = null;
+              const importModuleNamespace = (
+                loc: t.SourceLocation | null | undefined
+              ): t.Identifier => {
+                const _namespace = path.scope.generateUidIdentifierBasedOnNode(module);
+                imports.push(
+                  withLocation(
+                    requireTemplate({
+                      moduleStr: t.cloneNode(resolved),
+                      asId: t.cloneNode(_namespace),
```

Comment:
nit: I think it might be time to bypass `@babel/template` maybe? It's becoming increasingly easier to just write the raw AST, I think, which might be faster and not a lot of maintenance overhead, due to how small these snippets are